### PR TITLE
fixes target report subgroups and adds again the economic disadvantag…

### PR DESCRIPTION
**Pull request by esotrope**
…e filter to advanced filters and user group form

target report subgroups

image

advanced filters

image

user group form

image

### CODE REVIEW

File: webapp/src/main/resources/application.sql.yml [hunk 1]
```diff
@@ -44,11 +45,19 @@ sql:
...
+          g.code as grade_code
+          g.sequence as grade_order
+        FROM school s
+          JOIN (
+            SELECT DISTINCT
+              school_id,
+              grade_id
+            FROM exam
+          ) e ON e.school_id = s.id
+          JOIN grade g ON e.grade_id = g.id
```
Comment (esotrope) [inline webapp/src/main/resources/application.sql.yml]: unrelated sorting of sandbox roles

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html [hunk 2]
```diff
@@ -176,7 +178,7 @@
             [values]="filterOptions.genders"
             property="genders"
             enum="common.gender"
-            label="{{ 'common.filters.student.gender' | translate }}"
+            label="{{ 'common.student-field.Gender' | translate }}"
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html]: @malaffoon changed the label locations so there is a central location for where there were previously 3ish locations - do we need to add this to changelog? or is it not needed now that we have tenant/sandbox config?

File: webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts [hunk 3]
```diff
@@ -172,11 +173,9 @@ export class TargetReportComponent implements OnInit, ExportResults {
           name: this.translate.instant(
             `subject.${this.assessment.subject}.asmt-type.${
               this.assessment.type
-            }.level.${
-              this.subjectDefinition.performanceLevelStandardCutoff
-            }.name`
+            }.level.${this.subjectDefinition.overallScore.standardCutoff}.name`
           ),
-          id: this.subjectDefinition.performanceLevelStandardCutoff
+          id: this.subjectDefinition.overallScore.standardCutoff
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts]: unrelated - moved away from deprecated field

File: webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts [hunk 4]
```diff
@@ -493,47 +496,6 @@ export class TargetReportComponent implements OnInit, ExportResults {
     );
   }
 
-  private createAllSubgroups(settings: any): any[] {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/view/target-report/target-report.component.ts]: this was pretty extra so got rid of it

File: webapp/src/main/webapp/src/app/shared/filter/student-filter.ts [hunk 5]
```diff
@@ -20,44 +21,49 @@ export type ArrayFilter<T> = (
...
+    includes(filter.languages, student.languages) &&
+    includes(filter.militaryConnectedCodes, student.militaryConnectedCodes)
+  );
+}
+
 export function createStudentArrayFilter(
   filter: StudentFilter
 ): StudentArrayFilter {
-  const { isNullOrEmpty } = Utils;
-  return student => {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/filter/student-filter.ts]: made this more readable i hope ☝️

File: webapp/src/main/webapp/src/app/shared/form/component/information-icon/information-icon.component.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * This component is responsible for displaying a label with
+ * an information popover icon.
+ */
+@Component({
+  selector: 'app-info,[app-info]',
+  templateUrl: './information-icon.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InformationIconComponent {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/form/component/information-icon/information-icon.component.ts]: cloned `InformationButton` component but this one doesn't assume things like that you want the same popover title as the field label. also allows us to have icon buttons independent of field labels

File: webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html (hunk 2 above)
Comment (malaffoon) [reply webapp/src/main/webapp/src/app/assessments/filters/adv-filters/adv-filters.component.html]: we should put it in the change log with a couple examples of the keys that have changed

**Pull request merged**